elastic/beats
# [Auditbeat] Use a separate netlink socket for control to avoid data congestion.

## Proposed commit message
Use a separate socket for GetStatus.

## Checklist


## How to test this PR locally

Run this [program](https://gist.github.com/adriansr/16524e5b1242ce5cf59ca623a59a9ddb#file-bomb-go) while starting Auditbeat. In a separate terminal run `while true; do ss -f netlink | grep auditbeat; done`

You will see the auditbeat netlink sockets disappear but Auditbeat will continue to run.
Data will stop flowing to Elasticsearch, and an error message will be pushed Elasticsearch, which will be similar to this:
![image](https://github.com/user-attachments/assets/fab9a0f8-ada7-4708-9a92-7209407b076a)

With this PR's patch, the netlink sockets will remain, data will flow to Elasticsearch, and no error message will be pushed.
